smartsim/_core/_install/builder.py

```diff
@@ -317,6 +360,27 @@ def __init__(
         self.libtf_dir = libtf_dir
         self.torch_dir = torch_dir
 
+        # TODO: It might be worth making these constructor args so that users
+        #       of this class can configure exactly _what_ they are building.
+        os_ = platform.system().lower()
+        if "darwin" in os_:
+            self._os: TOperatingSystemStr = "darwin"
+        elif "linux" in os_:
+            self._os = "linux"
+        else:
+            raise BuildError(f"Unrecognized or unsupported operating system '{os_}'")
+
+        machine = platform.machine().lower()
+        if machine not in ("amd64", "x86_64") and any(
```

Comment:
Totally doable in this PR! I would object to a `dataclass` though as I'm not sure what it would mean to "instance" requirements. It seems to me we just want these in their own namespace. Maybe a number of `Enum`s?
```py
class AllowedOperatingSystem(enum.Enum):
    LINUX = ("linux", "linux2")
    DARWIN = ("darwin",)

    @classmethod
    def from_str(cls, string: str) -> "AllowedOperatingSystem":
        for enum_ in cls:
            if string in enum_.value:
                return enum_
        raise BuildError(f"Unrecognized or unsupported OS: {s}")

# Etc for architecture (and device?)
```
?

Reply:
That works for me. Let's restrict to OS and architecture for now. Also, I'm kind of vaguely wondering if htey actually need to be in the same class, since there's a potential future with Linux on arm64 or some weird combination that we might want to explicitly disallow

Reply:
That makes sense! I'll drop the "`Supported`/`Allowed`" wording as I think that that gives me the same anxiety as putting them in the same class and just make `Architecture` and `OperatingSystem` enums that we can use as types in place of the `t.Literal` strings that we are using now!